### ISSUE TITLE
cleos walletapi connection errors need improvement

### ISSUE THREAD

**Issue by sokoow** [EOSIO/eos#3892]
I get following when I try to connect to my wallet API through cleos:
```
cleos -n -v -u https://eos.XXX.YYY --wallet-url https://eos-wallet.XXX.YYY wallet list
Unable to connect to keosd, if keosd is running please kill the process and try again.
103359ms thread-0   main.cpp:2577                 main                 ] connect error: 1100000 connection_exception: Connection Exception
Unable to connect to keosd
    {}
    thread-0  main.cpp:726 try_port
```

It appears that cleos can connect to nodeos API through https no problem, but for some reason has problems with wallet API ? Not much debugging output there so...

**Comment by sokoow** [comment]
In https://github.com/EOSIO/eos/blob/379cb1a9e2a6557221d946eef4c5c83ddd156799/programs/cleos/main.cpp#L739

I see    ``` if (parsed_url.server != "localhost" && parsed_url.server != "127.0.0.1")
        return;``` - does that suppose to mean that cleos doesn't support remote keosd addresses, only localhost ?

**Comment by sokoow** [comment]
related to https://github.com/EOSIO/eos/issues/3878

**Comment by wanderingbort** [comment]
Closed as duplicate, will track progress in #3878
